src/lib/pickles/pickles.ml

```diff
@@ -333,7 +344,9 @@ module Make_str (_ : Wire_types.Concrete) = struct
                 =
              fun rule ->
               let (T (_, l)) = HT.length rule.prevs in
-              Vector.extend_exn (V.f l (M.f rule.prevs)) Max_proofs_verified.n 0
+              Vector.extend_front_exn
```

Comment:
It's a pity that we had to reverse all of these; probably there was just one place that was wrong before. (No need to change now though, since this works as written.)